### ISSUE TITLE
get error to move file to another branch: Failed to update branch ownership

### ISSUE THREAD

**Issue by yswtrue** [gitbutlerapp/gitbutler#3356]
get error to move file to another branch: Failed to update branch ownership
and can't commit get error: Failed to commit changes

**Comment by jokeyrhyme** [comment]
Yeah, I just started getting this error on 0.10.28 (I think, it's hard to figure out which version of GitButler I'm running)

**Comment by SiegfriedEhret** [comment]
Getting the same here.
I have a repo where I use pnpm and when I run "pnpm update", it updates the package.json and pnpm-lock.yaml file. I can move/commit the package.json file, but not the pnpm-lock.yaml one.

**Comment by y3owk1n** [comment]
Same issue here with "pnpm update", found out that if the changes on pnpm-lock is short enough, it will be able to do what it suppose to do, probably less than 100 lines of changes. For single repo, it will be easier to achieve this with single package update one by one, but for monorepo, it usually involves more files and changes, so it will will have the branch ownership or commit error.

**Comment by ozyx** [comment]
Same issue here. But the file is not super large, it is about 800 loc. There are 4 line changes. I can commit every single other file except this one.

**Comment by Byron** [comment]
Thanks for sharing, and sorry for the hassle. I myself also ran into the 'update ownership' issue, but couldn't determine what the problem actually was. In my case, a YAML file wasn't involved. As I moved on, the error seemed to have moved on as well.

It would definitely help if we could [collect the errors in the application logs](https://docs.gitbutler.com/development/debugging#logs) and share them here to maybe see a pattern emerge.

**Comment by krlvi** [comment]
I think this issue should be fixed now with the more recent releases. @yswtrue have you ran into this issue recently?

**Comment by yswtrue** [comment]
I can't use gitbutler anymore, because get this error
![CleanShot 2024-08-18 at 18 37 22@2x](https://github.com/user-attachments/assets/7a838855-6601-4dd0-9cba-f451797647b3)

**Comment by Byron** [comment]
It took me a moment to see the actual error in the far right of the screen, so I thought I'd repost it here:

<img width="502" alt="Screenshot 2024-08-18 at 19 46 59" src="https://github.com/user-attachments/assets/efc260d2-52c1-4fbb-894a-877890691775">

**Comment by estib-vega** [comment]
I got the original issue *"Failed to update branch ownership"* recently. 
I'm on version 0.12.23, macOS.
Will try to recreate

**Comment by Caleb-T-Owens** [comment]
We've entirely replaced the ownership mechanism with a more accurate approach. I'm going to close this since the original topic of the issue should be resolved.

If more errors come up with the new TreeChanges and assignments approach, we can open some new issues for this.

On the topic of the "No branches" error, I can't see any references to that error message in the codebase. If the "No branches" issue is still persisting, please open a new issue for it.
